design/cli-install-changes.md

```diff
@@ -0,0 +1,426 @@
+
+See discussion https://github.com/vmware-tanzu/velero/pull/2259#discussion_r384700723 for the two items below.
+
+- The `velero backup-location (create|set)` will have a new `--credentials mapStringString` flag which sets the name of the corresponding credentials secret for a provider. Format is provider:credentials-secret-name.
+
+- The `velero snapshot-location (create|set)` will have a new `--credentials mapStringString` flag which sets the list of name of the corresponding credentials secret for providers. Format is (provider1:credentials-secret-name1,provider2:credentials-secret-name2,...).
+
+Note that for this logic to work, either the plugin for which the secret corresponds to must have been installed in the system, or we must have a loop checking for when both are present before marking the BSL/VSL as ready.
+
+#### Examples of mounting secrets and environment variables
+
+With the changes proposed in the previous section (Credentials and secrets), the resulting deployment `yaml` would look like below:
+
+AWS
+```
+   spec:
+     containers:
+        volumeMounts:
+        - mountPath: /credentials
+          name: cloud-credentials-aws
+      - args:
+        - server
+        name: velero
+        env:
+        - name: AWS_SHARED_CREDENTIALS_FILE
+          value: /credentials/cloud
+     volumes:
+      - name: cloud-credentials
+        secret:
+          secretName: cloud-credentials
+```
+
+Azure
+```
+   spec:
+     containers:
+        volumeMounts:
+        - mountPath: /credentials
+          name: cloud-credentials-azure
+      - args:
+        - server
+        name: velero
+        env:
+        - name: AZURE_SHARED_CREDENTIALS_FILE
+          value: /credentials/cloud
+     volumes:
+      - name: cloud-credentials-azure
+        secret:
+          secretName: cloud-credentials-azure
+```
+
+
+#### Renaming "provider" to "plugin"
+
+As part of this change, we should change to use the term `plugin` instead of `provider`. The reasoning: in practice, we usually have 1 plugin per provider, and if there is an implementation for both object store and volume snapshotter for that provider, it will all be contained in the same plugin. When we handle plugins, we follow this logic. In other words, there's a plugin name (ex: `velero.io/aws`) and it can contain implementations of kind `ObjectStore` and/or `VolumeSnapshotter`.
+
+But when we handle BSL ir VSL (and the CLI commands/flags that configure them), we use the term `provider`, which can cause ambiguity as if that is a kind of thing different from a plugin. If the plugin is the "thing" that contains the implementation for the desired provider, we should make it easier for the user to guess that and change BackupStorageLocation/VolumeSnapshotLocation `Spec.Provider` field to be called `Spec.Plugin` and all related CLI command flags to `plugin`, and update the docs accordingly.
```

Comment:
Is `ir` meant to be `or`?

RE: changing the `Spec.Provider` -> `Spec.Plugin` field, I think that makes sense,  and will require a CRD version bump & deprecation cycle, but along with my previous comment, we should make the wording match across the board.

Reply:
Yes, `or`. I'll fix it.